flytepropeller/pkg/controller/nodes/catalog/datacatalog/transformer.go

```diff
@@ -114,9 +115,21 @@ func generateTaskSignatureHash(ctx context.Context, taskInterface core.TypedInte
 	return fmt.Sprintf("%v-%v", inputHashString, outputHashString), nil
 }
 
-// Generate a tag by hashing the input values
-func GenerateArtifactTagName(ctx context.Context, inputs *core.LiteralMap) (string, error) {
-	hashString, err := catalog.HashLiteralMap(ctx, inputs)
+// Generate a tag by hashing the input values which are not in cacheIgnoreInputVars
+func GenerateArtifactTagName(ctx context.Context, inputs *core.LiteralMap, cacheIgnoreInputVars *[]string) (string, error) {
+	var inputsAfterIgnore *core.LiteralMap
+	if cacheIgnoreInputVars != nil {
+		inputsAfterIgnore = &core.LiteralMap{Literals: make(map[string]*core.Literal)}
+		for name, literal := range inputs.Literals {
+			if slices.Contains(*cacheIgnoreInputVars, name) {
+				continue
+			}
+			inputsAfterIgnore.Literals[name] = literal
+		}
+	} else {
+		inputsAfterIgnore = inputs
+	}
```

Comment:
Rather than creating a new `LiteralMap` here would it make more sense to pass the `cacheIgnoreInputVars` to the `HashLiteralMap` function. Then in [this code here](https://github.com/flyteorg/flyte/blob/475e9d07644c612150f09598cf1b5eef1f1fa640/flyteplugins/go/tasks/pluginmachinery/catalog/hashing.go#L66-L68) we just need to update to:
```
	for name, literal := range literalMap.Literals {
	        if !slices.Contains(cacheIgnoreInputVars, name) {
		    hashifiedLiteralMap[name] = hashify(literal)
		}
	}
```
This achieves the same outcome in 2 LoC rather than 12. I think it makes more sense to only create a new LiteralMap in one place as well.